packages/marketplace/src/sagas/developer-detail.ts

```diff
@@ -0,0 +1,48 @@
+import { fetcher } from '@reapit/elements'
+import { call, takeLatest, put } from '@redux-saga/core/effects'
+import { errorThrownServer } from '../actions/error'
+import { logger } from 'logger'
+import {
+  developerDetailRequestDataFailure,
+  developerDetailReceiveData,
+  developerDetailLoading,
+} from '../actions/developer-detail'
+import errorMessages from '../constants/error-messages'
+import { URLS, generateHeader } from '../constants/api'
+import ActionTypes from '../constants/action-types'
+import { Action } from '../types/core'
+
+const INVALID_RESPONSE_ERROR = new Error('Invalid responses')
+export const developerDetailFetchDataFetch = function*({ data: { id } }: Action<{ id: string }>) {
+  yield put(developerDetailLoading(true))
+
+  try {
+    const developerDetailData = yield call(fetcher, {
+      url: `${URLS.developers}/${id}`,
+      method: 'GET',
+      api: window.reapit.config.marketplaceApiUrl,
+      headers: generateHeader(window.reapit.config.marketplaceApiKey),
+    })
+
+    if (developerDetailData) {
```

Comment:
`if (developerDetailData) {`
no need to check this if it no data it'll be jumping to catch block

Reply:
Let reuse fetchMyIdentity instead of write new